packages/godwoken/src/client.ts

```diff
@@ -172,7 +172,11 @@ export class GodwokenClient {
   public async getTransaction(
     hash: Hash
   ): Promise<L2TransactionWithStatus | undefined> {
-    return await this.rpcCall("get_transaction", hash);
+    const txWithStatus = await this.rpcCall("get_transaction", hash);
+    if (txWithStatus != null) {
+      return txWithStatus;
+    }
```

Comment:
Consider returning if `this.Rpc` points to the same instance with `this.readonlyRpc`.

Reply:
Updated